File: app/src/main/java/com/mxt/anitrend/adapter/recycler/shared/GroupTitleViewHolder.java
package com.mxt.anitrend.adapter.recycler.shared;

import android.view.View;

import com.mxt.anitrend.base.custom.recycler.RecyclerViewHolder;
import com.mxt.anitrend.databinding.AdapterEntityGroupBinding;
import com.mxt.anitrend.model.entity.group.RecyclerHeaderItem;
import com.mxt.anitrend.model.entity.group.RecyclerItem;

/**
 * Created by max on 2018/02/18.
 */

public class GroupTitleViewHolder extends RecyclerViewHolder<RecyclerItem> {

    private AdapterEntityGroupBinding binding;

    /**
     * Default constructor which includes binding with butter knife
     *
     * @param binding
     */
    public GroupTitleViewHolder(AdapterEntityGroupBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    @Override
    public void onBindViewHolder(RecyclerItem model) {
        binding.setModel((RecyclerHeaderItem) model);
        if (((RecyclerHeaderItem) model).getSize() < 1)
            binding.catalogHeaderCount.setVisibility(View.GONE);
        else
            binding.catalogHeaderCount.setVisibility(View.VISIBLE);
        binding.executePendingBindings();
    }

    @Override
    public void onViewRecycled() {
        binding.unbind();
    }

    @Override
    public void onClick(View v) {

    }

    @Override
    public boolean onLongClick(View v) {
        return false;
    }
}

File: app/src/main/java/com/mxt/anitrend/adapter/recycler/shared/GroupTitleViewHolder.kt
package com.mxt.anitrend.adapter.recycler.shared

import android.view.View
import com.mxt.anitrend.base.custom.recycler.RecyclerViewHolder
import com.mxt.anitrend.databinding.AdapterEntityGroupBinding
import com.mxt.anitrend.model.entity.group.RecyclerHeaderItem
import com.mxt.anitrend.model.entity.group.RecyclerItem

/**
 * Created by max on 2018/02/18.
 */
class GroupTitleViewHolder(
    private val binding: AdapterEntityGroupBinding,
) : RecyclerViewHolder<RecyclerItem>(binding.root) {
    override fun onBindViewHolder(model: RecyclerItem) {
        val header = model as? RecyclerHeaderItem ?: return
        binding.catalogHeaderTitle.text = header.getTitle()
        binding.catalogHeaderCount.text = header.size.toString()
        if (header.size < 1) {
            binding.catalogHeaderCount.visibility = View.GONE
        } else {
            binding.catalogHeaderCount.visibility = View.VISIBLE
        }
    }

    override fun onViewRecycled() {
    }

    override fun onClick(v: View) = Unit

    override fun onLongClick(v: View): Boolean = false
}
